### PR TITLE
INC-1228 Provide Find Prisoners on specified incentive level

### PR DESCRIPTION
**Pull request by Mjwillis**
New API end point:

`/prison/{prisonId}/incentive-level/{incentiveLevelCode}`

Which will provide a list of prisoners in a specific incentive level in a specific prison

### CODE REVIEW

**Review by andymarke — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchResource.kt [hunk 1]
```diff
@@ -93,4 +93,17 @@ class PrisonerSearchResource(private val prisonerSearchService: PrisonerSearchSe
     @ParameterObject @PageableDefault
     pageable: Pageable,
   ) = prisonerSearchService.findByPrison(prisonId.uppercase(), pageable, includeRestrictedPatients)
+
+  @GetMapping("/prison/{prisonId}/incentive-level/{incentiveLevelCode}")
+  @Operation(summary = "Get all prisoners in a prison, on a specified incentive level", description = "Requires ROLE_GLOBAL_SEARCH or ROLE_PRISONER_SEARCH role")
+  @Tag(name = "Batch")
+  @Tag(name = "Popular")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchResource.kt:100]: not sure `Popular` tag here is correct

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by andymarke — APPROVED** [review]
Lgtm

**Pull request merged**